aws/resource_aws_directory_service_directory.go

```diff
@@ -445,6 +465,9 @@ func resourceAwsDirectoryServiceDirectoryRead(d *schema.ResourceData, meta inter
 	if dir.Size != nil {
 		d.Set("size", *dir.Size)
 	}
+	if dir.Edition != nil {
```

Comment:
Normally this check wouldn't be necessary, however I think because we the SDK handles both SimpleAD and MicrosoftAD directory types that in fact we do want to keep this nil check in here. 👍